### PR TITLE
Newmodule yak

### PR DESCRIPTION
**Pull request by scorreard**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: .github/workflows/pytest-workflow.yml [hunk 1]
```diff
@@ -122,7 +122,8 @@ jobs:
             tags: subworkflows/bcl_demultiplex
           - profile: "conda"
             tags: subworkflows/vcf_annotate_ensemblvep
-
+          - profile: "conda"
+            tags: subworkflows/fasta_clean_fcs
```
Comment (mahesh-panchal) [inline .github/workflows/pytest-workflow.yml:126]: I think this shouldn't be here.

File: modules/nf-core/yak/main.nf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    yak \\
+        count \\
+        $args \\
+        -t${task.cpus} \\
+        -o ${prefix}.yak <(zcat $reads)
+
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        yak : \$(echo \$(yak --version 2>&1) | sed 's/^.*yak //' )
```
Comment (mahesh-panchal) [inline modules/nf-core/yak/main.nf:33]: You don't need this `echo`

Comment (scorreard) [reply modules/nf-core/yak/main.nf:33]:
```suggestion
        yak :  \$(yak --version 2>&1) | sed 's/^.*yak //' 
```

File: modules/nf-core/yak/main.nf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    yak \\
+        count \\
+        $args \\
+        -t${task.cpus} \\
+        -o ${prefix}.yak <(zcat $reads)
```
Comment (scorreard) [inline modules/nf-core/yak/main.nf:28]:
```suggestion
    if [[ $reads == *.gz ]]; then
        yak \\
            count \\
            $args \\
            -t${task.cpus} \\
            -o ${prefix}.yak <(gzip -dc $reads)
     else
        yak \\
            count \\
            $args \\
            -t${task.cpus} \\
            -o ${prefix}.yak < $reads
     fi            
```

File: modules/nf-core/yak/main.nf (hunk 2 above)
Comment (mahesh-panchal) [reply modules/nf-core/yak/main.nf:33]: The suggestion has the closing bracket in the wrong place.

File: modules/nf-core/yak/main.nf (hunk 3 above)
Comment (mahesh-panchal) [reply modules/nf-core/yak/main.nf:28]: To keep the script code shorter I usually do something like:
```nextflow
script:
def useGzip = !( assembly instanceof List ? assembly.every{ it.name.endsWith(".gz") } : assembly.name.endsWith(".gz") )
"""
${useGzip ? 'gzip -c' : 'cat'} $assembly >output.gz
"""
```
The idea is a test is written in groovy, and then the value of it is used in an expression inside the script code to determine which command to use.

**Comment by scorreard** [comment]
Closing this PR without merging as a lot of changes were made to module creation and nf-test, I think it would be better to strat from scratch.

**Pull request closed without merging**